media/js/src/AssetFilter.jsx

```diff
@@ -6,17 +6,80 @@ export default class AssetFilter extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            titleFilter: '',
             selectedOwner: 'all'
         };
-        this.handleChange = this.handleChange.bind(this);
     }
-    handleChange(e) {
-        console.log('handleChange');
+    handleTitleSearch(e) {
+        if (e.key === 'Enter') {
+            this.props.handleTitleFilterSearch();
+        }
+    }
+    handleOwnerChange(e) {
+        this.setState({selectedOwner: e.value});
     }
     render() {
         const { selectedOwner } = this.state;
 
+        let ownersOptions = [];
```

Comment:
Because these are working with only props, it might make sense to move this to the constructor so it doesn't run with each call to render.

Reply:
This does need to be updated whenever the props are updated, though.. moving this to the constructor causes the options to not populate, since the assets are loaded over XHR - they start out as null. It does seem like a lot of work to do on each render - i will work on cleaning this up.